templates/layout/mainnavigation.html

```diff
@@ -55,14 +55,27 @@
 {{ define "mainNavigationItem" }}
   {{ if not .IsHidden }}
     <a class="dropdown-item" href="{{ .Path }}">
-      <span class="nav-icon">
-        {{ if .CustomIcon }}
-          {{ includeSvg .CustomIcon }}
-        {{ else }}
-          <i class="fas {{ .Icon }} mr-2"></i>
-        {{ end }}
-      </span>
-      <span class="nav-text">{{ .Label }}</span>
+      {{ if .IsHighlighted }}
+        <span class="btn btn-primary">
+          <span class="nav-icon">
+            {{ if .CustomIcon }}
+              {{ includeSvg .CustomIcon }}
+            {{ else }}
+              <i class="fas {{ .Icon }} mr-2"></i>
+            {{ end }}
+          </span>
+          <span class="text-white">{{ .Label }}</span>
+        </span>
+      {{ else }}
+        <span class="nav-icon">
+          {{ if .CustomIcon }}
+            {{ includeSvg .CustomIcon }}
+          {{ else }}
+            <i class="fas {{ .Icon }} mr-2"></i>
+          {{ end }}
+        </span>
+        <span class="nav-text">{{ .Label }}</span>
+      {{ end }}
```

Comment:
This creates duplicated code, consider something like this:
```suggestion
      {{ $buttonClass := "" }}
      {{ $textClass := "nav-text" }}
      {{ if .IsHighlighted }}
        {{ $buttonClass = "btn btn-primary" }}
        {{ $textClass = "text-white" }}
      {{ end }}


      <span class="{{ $buttonClass }}">
        <span class="nav-icon">
          {{ if .CustomIcon }}
            {{ includeSvg .CustomIcon }}
          {{ else }}
            <i class="fas {{ .Icon }} mr-2"></i>
          {{ end }}
        </span>
        <span class="{{ $textClass }}">{{ .Label }}</span>
      </span>
```

Reply:
Thanks, I did not consider that you could create a `span` with an empty `class`.

Reply:
I think the horizontal alignment is supposed to be this way. The button starts left aligned and then wants a bit of space until the icon starts. See the screenshot of the mobile version in the BIDS.
I will check if the vertical alignment can be improved.